altair/vegalite/v5/api.py

```diff
@@ -2988,6 +3040,32 @@ def __or__(self, other):
         copy |= other
         return copy
 
+    def _transformed_data(
+        self,
+        row_limit: Optional[int] = None,
+        exclude: Optional[Iterable[str]] = None,
+    ) -> List[pd.DataFrame]:
```

Comment:
Output is always a list of pandas dataframes? Do we want this to eventually support other dataframes-a-like as well?

Reply:
Right now in VegaFusion, these results can actually be an Arrow tables (if the input Chart(s) wrap arrow tables), a Polars DataFrames (if the input Chart(s) wrap Polars DataFrames), or pandas DataFrames (all other cases).

It seemed to me that this would make for a pretty awkward type signature, so I left it as just pandas.  It could be a `Union`, but arrow and polars are optional dependencies so I'm not sure how to have the Union include different options based on what's installed.  Do you have any ideas here @mattijn or @binste?

Reply:
Can we define a `DataFrameLike` type (elsewhere in the codebase) that can refer to any object with a `.__dataframe__` attribute, regardless of whether it's a pandas DataFrame, polars DataFrame, or any other object? And use that type here? Would that help?

Reply:
That's a good idea, I'll look into that.

Reply:
Done in 6f43d6b94614134792671d2c1a90aff588682c94

Reply:
Can you double check this @binste? Should a type definition comes back in the root of altair (see changes in `__init__`)?

Reply:
I'm in favor of renaming it to `_DataFrameLike` so no one starts using it themselves and it won't show up in `__init__`. I'm doing the same in #2976. Gives us the flexibility to gather these types at a later stage into a `types.py` module if that makes sense or move them around/redefine otherwise.

Apart from that I like the protocol usage. We can narrow down the type further once we figure out how to best do that but at least it's better then `Any` while being correct.

Reply:
Could you also move the `Protocol` import into a `if sys.version_info >= (3, 8):` statement? It was introduced in Python 3.8 and we try to import from the official library as soon as it's available.

Reply:
Yes, I'll move the protocol. 

I might not have the right understanding here, but since `DataFrameLike` is in the return type of the public method, doesn't the type need to be public as well? Otherwise, Altair users wouldn't be able to work with it in their own typed code.